pykube/objects.py

```diff
@@ -450,98 +437,157 @@ def ready(self):
 
 
 class ReplicaSet(NamespacedAPIObject, ReplicatedMixin, ScalableMixin):
-
     version = "apps/v1"
     endpoint = "replicasets"
     kind = "ReplicaSet"
 
 
 class Secret(NamespacedAPIObject):
-
     version = "v1"
     endpoint = "secrets"
     kind = "Secret"
 
 
 class Service(NamespacedAPIObject):
-
     version = "v1"
     endpoint = "services"
     kind = "Service"
 
+    def proxy_http_request(self, method, path, **kwargs) -> Response:
+        """Issue a HTTP request with specific HTTP method to proxy of a Service.
+        Args:
+            :param method: The http request method e.g. 'GET', 'POST' etc.
+            :param path: The URI path for the request.
+            :param kwargs: Keyword arguments for the proxy_http_get function.
+            They are the same as for requests.models.Request object
+            plus the additional 'port' kwarg, which can be used to override the
+            default (first defined) port used to connect to the Service.
+        Returns:
+            The requests.Response object.
+        """
+        if "port" in kwargs:
```

Comment:
please make `port` an optional argument (`port=None`) and add type annotations for parameters